src/mpid/ch4/netmod/ucx/ucx_progress.h

```diff
@@ -8,6 +8,42 @@
 
 #include "ucx_impl.h"
 
+MPL_STATIC_INLINE_PREFIX int MPIDI_UCX_handle_deferred_ops(void)
+{
+
+    int mpi_errno = MPI_SUCCESS;
+    MPIDI_UCX_deferred_am_isend_req_t *dreq = MPIDI_UCX_global.deferred_am_isend_q;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_UCX_HANDLE_DEFERRED_OPS);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_UCX_HANDLE_DEFERRED_OPS);
+
+    if (dreq) {
```

Comment:
So does this only issue one segment per execution of the progress loop? That seems in efficient...I operate from the perspective that once we hold the critical section, we should do as much work as possible to avoid the overhead of repeated lock/unlock.